doc/library/d3viz/index.txt

```diff
@@ -130,6 +130,19 @@ may not be perfect. In this case, you can press the ``Release node``
 button in the top-left corner to automatically arrange nodes. To reset
 nodes to their default position, press the ``Reset nodes`` button.
 
+You can also display the interactive graph inline in
+IPython using ``IPython.display.IFrame``:
+
+.. code:: python
+
+    from IPython.display import IFrame
+    d3v.d3viz(predict, 'examples/mlp.html')
+    Iframe('examples/mlp.html', width=700, height=500)
```

Comment:
thanks for checking. I'll merge this PR when travis pass.

On Thu, May 12, 2016 at 3:20 PM, Mikhail Korobov <notifications@github.com>
wrote:

> In doc/library/d3viz/index.txt
> <https://github.com/Theano/Theano/pull/4505#discussion_r63080612>:
>
> > @@ -130,6 +130,19 @@ may not be perfect. In this case, you can press the ``Release node``
> >  button in the top-left corner to automatically arrange nodes. To reset
> >  nodes to their default position, press the ``Reset nodes`` button.
> >
> > +You can also display the interactive graph inline in
> > +IPython using ``IPython.display.IFrame``:
> > +
> > +.. code:: python
> > +
> > +    from IPython.display import IFrame
> > +    d3v.d3viz(predict, 'examples/mlp.html')
> > +    Iframe('examples/mlp.html', width=700, height=500)
>
> I believe it should be possible to implement that; the tricky part is to
> make sure related resources are available, but it should be doable. It will
> likely require a custom IPython extension which users will have to
> %load_ext though (similar to https://github.com/jovyan/pythreejs). I'm
> interested in working on that, but it probably doesn't belong to this
> ticket. Theano IPython extension can do more cool things than just
> providing js/css fils for d3viz.
>
> I think embedding JS and CSS in a html string won't work because IPython
> sanitizes HTML displayed via display.HTML. Just did a quick prototype -
> d3viz no longer works after embedding all scripts and styles when displayed
> with display.HTML.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/4505/files/d43c6ad163be5aa24229c02beaadedba0306bc25..5494d94a57e3ee6aa115640e78aabbeab2f0a0cb#r63080612>
>